vpc-endpoints.tf

```
@@ -12,6 +26,7 @@ resource "aws_vpc_endpoint" "s3" {
 
   vpc_id       = local.vpc_id
   service_name = data.aws_vpc_endpoint_service.s3[0].service_name
+  policy       = var.s3_endpoint_policy == "" ? data.aws_iam_policy_document.default.json : var.s3_endpoint_policy
```

I haven't tested myself yet, but I think something like this should work without requiring the default policy data source:
```suggestion
  policy       = var.s3_endpoint_policy == "" ? null : var.s3_endpoint_policy
```

---

one other thing that might work as well (again, haven't tested this here) would be:
1. Change the policy default value to `null` like:
```hcl
variable "s3_endpoint_policy" {
  description = "Custom IAM policy for S3 endpoint"
  type        = string
  default     = null
}
```
2. Then simply do this:
```hcl
  policy       = var.s3_endpoint_policy
```

again, it would need to be tested and validated on either approach but I think it might work and makes it cleaner. I don't know what the AWS provider or AWS API will provide when a null is provided (or no policy is provided)